### DISCUSSION TITLE
Is Neuropixel 2 4-shank supported in CellExplorer?

### DISCUSSION THREAD

**Discussion by taningh86** [petersenpeter/CellExplorer#113] [General]
Hi,
I have just started to use cellExplorer and trying to figure things out. I am assuming I will have more questions in the future and I hope that's ok. Will very much appreciate your help in this process. 
The first question I have for now is if 4-shank neuropixels probe is supported in CellExplorer? I only see IMEC NPX 1.0. But if it is possible somehow then how to add the electrode groups. Is there a guide to it? I am not seeing in the documentations. Please let me know.

Thanks
Jimmy

**Comment by petersenpeter** [comment]
Hi Jimmy

Indeed, CellExplorer supports Neuropixels and most other probes. It does not have the probe layout built-in, unfortunately, and it cannot read that from other formats. However, it would not be a lot of work to get it to do so. You need to create electrode groups in your session struct, and that information you either have to create manually, or extract it from other metadata from e.g. Open Ephys. 

To create the electrode groups (e.g. from two shanks), you can do something along:
```m
session.extracellular.electrodeGroups.channels{1} = [1:96];
session.extracellular.electrodeGroups.channels{2} = [97:192];
```

Now to create the layout, you can use the built-in method from the session GUI or define it as below: 
```m
session.extracellular.chanCoords.x(1:92)= 0;
session.extracellular.chanCoords.x(97:192)= 200;
session.extracellular.chanCoords.y(1:92)= 0:10:910;
session.extracellular.chanCoords.y(97:192)= 0:10:910;
```

There is also a [video](https://www.youtube.com/watch?v=ejI5VIz9Yw8&embeds_referring_euri=https%3A%2F%2Fcellexplorer.org%2F&source_ve_path=MjM4NTE) and [tutorial](https://cellexplorer.org/tutorials/neuropixels-tutorial/) covering Neuropixel 1 probes.

**Comment by taningh86** [comment]
Hi Peter. Thank you so much for the input. Its a 4 shank probe, so I would think the logic you gave for 2 shanks should also apply to 4 shanks I believe? Also, I am assuming 0, 200, 0:10:910 are measurements of the probe x and Y distances?

**Comment by petersenpeter** [comment]
Yes to both questions -  it was just to give you a hint as to how to get started...

**Comment by taningh86** [comment]
Thank you. I will work on it and reach out if I have any questions.
